packages/runtime/container-runtime/src/gc/gcDefinitions.ts

```diff
@@ -218,6 +219,45 @@ export const GCNodeType = {
  */
 export type GCNodeType = (typeof GCNodeType)[keyof typeof GCNodeType];
 
+/**
+ * The contents of a garbage collection sweep message.
+ * @internal
+ */
+export interface ISweepMessageContents {
+	// The ids of nodes that are deleted.
+	deletedNodeIds: string[];
+}
+
+/**
+ * The content type of a garbage collection message content.
+ * @internal
+ */
+export type GarbageCollectionMessageContents = ISweepMessageContents;
+
+/**
+ * The const types of a garbage collection message.
+ * @internal
+ */
+export const GarbageCollectionMessageType = {
+	// Message that is sent after sweep runs.
+	Sweep: "Sweep",
+};
+/**
+ * The types of a garbage collection message.
+ * @internal
+ */
+export type GarbageCollectionMessageType =
+	(typeof GarbageCollectionMessageType)[keyof typeof GarbageCollectionMessageType];
+
+/**
+ * Interface for a message to be used for sending / received garbage collection messages.
+ * @internal
+ */
+export interface IGarbageCollectionMessage {
```

Comment:
This doesn't quite look right to me. If there were 2 types it wouldn't coordinate the `type` and `contents`'s type. An alternative is to do `{ type: "Sweep"; contents: ISweepMessageContents }`, and you'd `|` together as many of those as there are `type`s as more are added.

Also then you can simply do `export type GarbageCollectionMessageType = IGarbageCollectionMessage["type"];`

If you want the `cosnt GarbageCollectionMessageType` as well you can, I'm not sure pros/cons.

Reply:
[TS Playground](https://www.typescriptlang.org/play?#code/MYewdgzgLgBANiA5jAvDUkRwKYDoGIDcAUMVAJ4AO2MAkgMoDu22lAsthBAIaLYDC4KNjBQIqGAG8YAMxAgAXDDABXALYAjbACcYAXxIVqdACohN0cNg5deAoSLETpG7tqWrNO-STJUatADibq58gnA4wFAAluA2PHwSxAA+UsQwGTBG2EoAREwslLkkmegOohBKDMys8XaCoo4QxHopaaXZeWYWUFbF6ZkYwhVV3RqWYNacCfaNFS3Evtkwwdqh9hHYUbFgdXwm-hJBIfVYkTFx03YA2rnZuQC6JEA)

Reply:
Updated it.